torch_geometric/explain/algorithm/subgraphX_explainer.py

```diff
@@ -0,0 +1,646 @@
+import copy
+import logging
+import math
+from collections import Counter
+from functools import partial
+from os.path import isfile, join
+from typing import Callable, Dict, List, Optional, Union
+
+import networkx as nx
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Batch, Data
+from torch_geometric.explain import Explanation
+from torch_geometric.explain.config import ModelTaskLevel
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.typing import EdgeType, NodeType
+from torch_geometric.utils import (
+    k_hop_subgraph,
+    remove_self_loops,
+    to_networkx,
+)
+
+from .base import ExplainerAlgorithm
+from .subgraphX_utils import (
+    GnnNetsGC2valueFunc,
+    GnnNetsNC2valueFunc,
+    find_closest_node_result,
+    gnn_score,
+    reward_func,
+    sparsity,
+)
+
+
+def compute_scores(score_func, children):
```

Comment:
Could you add docstrings and types to these functions to explain their objective?